verifcid/validator.go

```diff
@@ -0,0 +1,54 @@
+package verifcid
+
+import (
+	"github.com/ipfs/go-cid"
+	mh "github.com/multiformats/go-multihash"
+)
+
+type Allowlist interface {
```

Comment:
I feel like it should be moved to go-cid instead. Although, it's not that widely used in PL.